zulipterminal/ui_tools/boxes.py

```diff
@@ -57,6 +59,50 @@ def main_view(self, new: bool) -> Any:
     def set_editor_mode(self) -> None:
         self.view.controller.enter_editor_mode_with(self)
 
+    def set_text_and_refresh(self, text_box: urwid.Text, text: str) -> None:
+        # Timer displays on the screen would need
+        # immediate screen refreshes to display changes accurately.
+        text_box.set_text(text)
+        self.view.controller.update_screen()
+
+    @asynch
+    def edit_countdown_timer(self, timer_box: urwid.Text,
+                             message_timestamp: float,
+                             msg_content_edit_limit: int,
+                             *, narrow_type: Literal['stream', 'private']
+                             ) -> None:
+        edit_deadline = msg_content_edit_limit + message_timestamp
+
+        while edit_deadline > time():
+            seconds_left = edit_deadline - time()
+            minutes = int(seconds_left // 60)
+            seconds = seconds_left % 60
+            if minutes >= 1:
+                self.set_text_and_refresh(
+                    timer_box,
+                    "{} min to edit".format(minutes)
+                    )
+                if seconds > 0:
+                    # Sleep for the variable time between the initiation of
+                    # the timer to the next minute.
+                    sleep(seconds)
+                else:
+                    # Sleep for a complete minute before updating timer.
+                    sleep(60)
+            elif seconds >= 1:
+                self.set_text_and_refresh(
+                    timer_box,
+                    "{} sec to edit".format(math.floor(seconds))
+                    )
+                # Update the timer every second when the time remaining
+                # is less than 1 minute.
+                sleep(1)
+
+        if narrow_type == "stream":
+            self.set_text_and_refresh(timer_box, "Topic-editing only")
+        else:
+            self.set_text_and_refresh(timer_box, "Time's up!")
```

Comment:
Yes, I did consider calling it `Edit time's up!` but went with this because I wanted it to be consistent with the webapp. I'll make it more specific. The warning sounds useful and considerate towards users, I'll add that in as well.